packages/store/addon/-private/system/model/internal-model.ts

```diff
@@ -55,14 +64,27 @@ function relationshipStateFor(instance: InternalModel, propertyName: string) {
 
 const { hasOwnProperty } = Object.prototype;
 
-/**
-  @module @ember-data/store
-*/
-
-// move to TS hacks module that we can delete when this is no longer a necessary recast
-type ManyArray = InstanceType<typeof ManyArray>;
-type PromiseBelongsTo = InstanceType<typeof PromiseBelongsTo>;
-type PromiseManyArray = InstanceType<typeof PromiseManyArray>;
+let ManyArray: ManyArray;
+let PromiseBelongsTo: PromiseBelongsTo;
+let PromiseManyArray: PromiseManyArray;
+
+let _found = false;
+let _getModelPackage: () => boolean;
+if (HAS_MODEL_PACKAGE) {
+  _getModelPackage = function() {
+    if (!_found) {
+      let modelPackage = require('@ember-data/model/-private');
+      ({ ManyArray, PromiseBelongsTo, PromiseManyArray } = modelPackage);
+      if (ManyArray && PromiseBelongsTo && PromiseManyArray) {
+        _found = true;
+
+        // Some tests require PromiseManyArray instanceof PromiseArray to be true
+        Object.setPrototypeOf(PromiseManyArray, PromiseArray);
```

Comment:
if we're doing this hack here to fix tests then something unexpected has changed which we should fix